frappe/build.py

```diff
@@ -11,24 +11,141 @@
 import tempfile
 from distutils.spawn import find_executable
 
-from six import iteritems, text_type
-
 import frappe
 from frappe.utils.minify import JavascriptMinify
 
+import click
+from requests import get
+from six import iteritems, text_type
+from six.moves.urllib.parse import urlparse
+
 
 timestamps = {}
 app_paths = None
+sites_path = os.path.abspath(os.getcwd())
+
+
+def download_file(url, prefix):
+	filename = urlparse(url).path.split("/")[-1]
+	local_filename = os.path.join(prefix, filename)
+	with get(url, stream=True, allow_redirects=True) as r:
+		r.raise_for_status()
+		with open(local_filename, "wb") as f:
+			for chunk in r.iter_content(chunk_size=8192):
+				f.write(chunk)
+	return local_filename
+
+
+def build_missing_files():
+	# check which files dont exist yet from the build.json and tell build.js to build only those!
+	missing_assets = []
+	current_asset_files = []
+
+	for type in ["css", "js"]:
+		current_asset_files.extend(
+			[
+				"{0}/{1}".format(type, name)
+				for name in os.listdir(os.path.join(sites_path, "assets", type))
+			]
+		)
+
+	with open(os.path.join(sites_path, "assets", "frappe", "build.json")) as f:
+		all_asset_files = json.load(f).keys()
+
+	for asset in all_asset_files:
+		if asset.replace("concat:", "") not in current_asset_files:
+			missing_assets.append(asset)
+
+	if missing_assets:
+		from subprocess import check_call
+		from shlex import split
+
+		click.secho("\nBuilding missing assets...\n", fg="yellow")
+		command = split(
+			"node rollup/build.js --files {0} --no-concat".format(",".join(missing_assets))
+		)
+		check_call(command, cwd=os.path.join("..", "apps", "frappe"))
+
+
+def get_assets_link(frappe_head):
+	from subprocess import getoutput
+	from requests import head
+
+	tag = getoutput(
+			"cd ../apps/frappe && git show-ref --tags -d | grep %s | sed -e 's,.*"
+			" refs/tags/,,' -e 's/\^{}//'"
+			% frappe_head
+		)
+
+	if tag:
+		# if tag exists, download assets from github release
+		url = "https://github.com/frappe/frappe/releases/download/{0}/assets.tar.gz".format(tag)
```

Comment:
Should we make this URL configurable?

Reply:
A similar thought came to mind while I wrote this. I couldn't chalk up any particular use case for that though as this feature only exists for the Frappe's assets alone. 

But if a valid use case exists, we should consider supporting it.

Reply:
Yeah, anyway we can add it later if we find the need